django/db/models/sql/query.py

```diff
@@ -650,13 +650,13 @@ def exists(self, limit=True):
                 for combined_query in q.combined_queries
             )
         q.clear_ordering(force=True)
-        if limit:
+        if limit is True:
```

Comment:
Is this idiomatic for how we should be checking boolean values? I see there is also `q.group_by is True` in this method, but this seems unusual.

Reply:
I'm not sure how idiomatic it is in Django's codebase, or in Python in general.

I've changed the comparison from truthy to identity in order to have a stricter checking. See https://code.djangoproject.com/ticket/35585#comment:4

Reply:
Sorry I didn't read the ticket carefully. It seems like an overreaction to change this and add a test, especially considering the mistake did not cause any bugs. Just my opinion!

Reply:
so what is the consensus?

* test y/n?
* identity check y/n?

Reply:
I have read the comments, I'm of the opinion that test(s) are needed, but I will review this in detail tomorrow.
I have ran out of time today because of the "juicy" security release that was just issued. I'll post an update soon!

Reply:
I haven't forgotten about this pending resolution. I'm currently considering the next steps for this PR, especially in light of recent comments/reactions that have given me pause.

Reply:
I wouldn't sweat this too much. Just remove `using` and move on. This sort of mistake isn't likely and didn't cause a bug.

Reply:
After further consideration, for the sake of simplicity and pragmatism, I have removed the test using mock. I'll proceed with merging once CI is green.